crypto/x509/x509_set.c

```diff
@@ -238,3 +240,72 @@ const X509_ALGOR *X509_get0_tbs_sigalg(const X509 *x) {
 X509_PUBKEY *X509_get_X509_PUBKEY(const X509 *x509) {
   return x509->cert_info->key;
 }
+
+static int X509_SIG_INFO_get(const X509_SIG_INFO *sig_info, int *digest_nid,
+                             int *pubkey_nid, int *sec_bits, uint32_t *flags) {
+  if (digest_nid != NULL) {
```

Comment:
do we need a null check for `sig_info`?

Reply:
This is an external function in OpenSSL that we might end up exposing some day, so sure.